src/lumigo_opentelemetry/external/botocore/extensions/sqs.py

```diff
@@ -77,34 +77,56 @@ class ContextableList(list):
         def __init__(self, messages, span: Span):
             super().__init__(messages)
             self._span = span
-            self._iteration_ongoing = False
+            self._scope_depth = 0
             self._token = None
 
+        # Iterator support
         def __getitem__(self, key: int) -> Any:
             try:
                 return super(_SqsExtension.ContextableList, self).__getitem__(key)
             finally:
-                # NonRecordingSpan is what we get when there is no span in the context
-                if self._iteration_ongoing and isinstance(get_current_span(), NonRecordingSpan) and self._span:
-                    # If there is not current active span (e.g. in case of an overarching entry span),
-                    # we restore as active the "ReceiveMessage" span. We neither record exceptions on it
-                    # nor set statuses, as the "ReceiveMessage" span is already ended when it is used as
-                    # parent by spans created while iterating on the messages.
-                    self._token = context_api.attach(context_api.set_value(_SPAN_KEY, self._span))
+                pass
+                # self._enter_scope()
```

Comment:
don't leave commented-out code
and then you can drop this `finally` block

Reply:
Yes. The entire method can go actually :D

Reply:
Done in ce7e4d51683442078625586fffcc48e86af660d6